src/runtime/pipeline/pipeline_executor.h

```diff
@@ -0,0 +1,183 @@
+   *  by storage_id. If not given, linked parameters are looked-up using an internal implementation,
+   *  which is not compatible with RPCModules.
+   */
+  void Init(const Array<tvm::runtime::Module>& modules, const std::string& pipeline_json);
+
+  /*!
+   * \brief set index-th input to the graph.
+   * \param index The input index.
+   * \param data_in The input data.
+   */
+  void SetInput(int index, DLTensor* data_in, int modIndx);
+
+  /*!
+   * \brief get index-th input.
+   * \param index The input index.
+   * \return The input data.
+   */
+  NDArray GetInput(int index, int mIndx) const;
+
+  /*!
+   * \brief get input index-th by name.
+   * \param input name.
+   * \return The input index.
+   */
+  int GetInputIndex(const string& name, int mIndx) const;
+  /*!
+   * \brief Get the number of outputs
+   *
+   * \return The number of outputs from graph.
+   */
+  int NumOutputs() const;
+  /*!
+   * \brief Get the number of inputs
+   *
+   * \return The number of inputs to the graph.
+   */
+  int NumInputs() const;
+  /*!
+   * \brief Return NDArray Array for all output.
+   *
+   * \param syncPoll Syncholization poll mode or ASyncholization.
+   * \return NDArray Array for all output.
+   */
+  Array<NDArray> GetOutput(bool syncPoll = true);
+
+  void Load(dmlc::JSONReader* reader) {
+    reader->BeginArray();
+    while (reader->NextArrayItem()) {
+      std::string key;
+      reader->BeginObject();
+      int mod_indx = 0;
+      unordered_map<int, unordered_map<int, string>> output;
+      while (reader->NextObjectItem(&key)) {
+        if (key == "mod_indx") {
+          reader->Read(&mod_indx);
+        }
+        if (key == "output") {
+          reader->BeginArray();
+          while (reader->NextArrayItem()) {
+            int output_indx = -1;
+            unordered_map<int, string> depend;
+            reader->BeginObject();
+            while (reader->NextObjectItem(&key)) {
+              if (key == "output_indx") {
+                reader->Read(&output_indx);
+              }
+              if (key == "dependent") {
+                reader->BeginArray();
+                int dep_mod_indx = -1;
+                string inputName;
+                while (reader->NextArrayItem()) {
+                  reader->BeginObject();
+                  while (reader->NextObjectItem(&key)) {
+                    if (key == "mod_indx") {
+                      reader->Read(&dep_mod_indx);
+                    }
+                    if (key == "input_name") {
+                      reader->Read(&inputName);
+                    }
+                  }
+                  if (dep_mod_indx >= 0) {
+                    depend[dep_mod_indx] = inputName;
+                  }
+                }
+              }
+            }
+
+            if (output_indx >= 0) {
+              output[output_indx] = depend;
+            }
+          }
+        }
+      }
+      if (mod_indx >= 0) {
+        pipeline_conf[mod_indx] = output;
+      }
+    }
```

Comment:
This logic needs to be improved. It's impossible to be maintained in the future.

Reply:
fixed.